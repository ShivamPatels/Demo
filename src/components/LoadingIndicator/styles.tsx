import { StyleSheet } from "react-native";
import { screenHeight, screenWidth } from "../../utils/dimensions";

const styles = StyleSheet.create({
    container: {
        position: "absolute",
        width: screenWidth,
        height: screenHeight,
        backgroundColor: "transparent",
        zIndex: 1,
        elevation: 2,
    },
    middleView: {
        flex: 1,
        justifyContent: "center",
        alignItems: "center",
    },
});
export default styles;
