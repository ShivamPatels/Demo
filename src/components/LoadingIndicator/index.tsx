import React from "react";
import { View, ActivityIndicator } from "react-native";
import styles from "./styles";

const LoadingIndicator = () => {
    return (
        <View style={styles.container}>
            <View style={styles.middleView}>
                <ActivityIndicator size="large" />
            </View>
        </View>
    );
};

export default LoadingIndicator;
