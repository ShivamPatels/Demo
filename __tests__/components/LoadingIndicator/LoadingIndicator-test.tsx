/**
 * @format
 */

import "react-native";
import React from "react";
import renderer from "react-test-renderer";
import LoadingIndicator from "../../../src/components/LoadingIndicator";

it("renders correctly", () => {
    const tree = renderer.create(<LoadingIndicator />).toJSON();
    expect(tree).toMatchSnapshot();
});
