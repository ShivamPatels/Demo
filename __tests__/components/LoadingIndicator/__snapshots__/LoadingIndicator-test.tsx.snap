// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View
  style={
    Object {
      "backgroundColor": "transparent",
      "elevation": 2,
      "height": 1334,
      "position": "absolute",
      "width": 750,
      "zIndex": 1,
    }
  }
>
  <View
    style={
      Object {
        "alignItems": "center",
        "flex": 1,
        "justifyContent": "center",
      }
    }
  >
    <ActivityIndicator
      size="large"
    />
  </View>
</View>
`;
